src/test/clojure/jarl/compliance_test.clj

```diff
@@ -56,28 +81,19 @@
                 (get test-case "input"))
         info (cond-> (parse ir compliance-builtin-resolver)
                      (true? (get test-case "strict_error")) (assoc :strict-builtin-errors true))]
-    (doseq [entry-point entry-points]
-      (let [want-result (get want-results entry-point)
-            plan (get-plan (get info :plans) entry-point)]
-        (is (not (nil? plan)))
-        (when-not (nil? plan)
-          (if (and (nil? want-error-code) (nil? want-error))
-            (let [result-set (plan info data input)
-                  result (get (first result-set) "result")]
-              ;(println (str "Want: " want-result "\n\nGot: " result "\n"))
-              (is (= result want-result)))
-            (do
-              (log/infof "Want error: %s: %s" want-error-code want-error)
-              (try
-                (let [result-set (plan info data input)]
-                  (is (nil? result-set) (str "Exception must be thrown, got " result-set)))
-                (catch JarlException e
-                  (is (.contains (.getMessage e) want-error) (str "Got error message: " (ex-message e)))
-                  (is (= (.getType e) want-error-code) (str "Got error code: " (.getType e))))
-                (catch Throwable e
-                  (do
-                    (println "Unexpected error" e)
-                    (is false "JarlException must be thrown")))))))))))
+    (if (and (nil? want-error-code) (nil? want-error))
+      (let [result (eval-entry-points-for-results info entry-points data input)]
+        (is (= result want-result)))
+      (let [errors (eval-entry-points-for-errors info entry-points data input)]
+        ; There might be other errors generated than what is expected by the test case definition, but the test case
+        ; doesn't know we're executing multiple entry-points, so we can't count unexpected JarlExceptions as violations
+        (is (>= (count (filter
```

Comment:
`(is (not-empty (filter ...))` reads more natural here, or WDYT?
Might also be worth considering using a binding for the `(filter ..)` expression in the `let` above, making the `is` more radable, i.e:

```clojure
(is (not-empty jarl-exceptions) 
    (str "Expected error code:" want-error-code "; message: " want-error ", got" errors)))
```